SystemManager/Inc/SM.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * SM.hpp
+ *
+ * System Manager Header
+ *
+ * Created on: Nov 7, 2022
+ * Author(s): Gordon Fountain
+ */
+
+#ifndef ZPSW3_SM_HPP
+#define ZPSW3_SM_HPP
+
+#include <stdint.h>
+#include "cmsis_os.h"
+// #include "LOS_Link.hpp"
+// #include "LOS_Actuators.hpp"
+#include "SM_StateManager.hpp"
+
+class SystemState;
+
+namespace SM {
+    // Enumerates the current drone status
+    enum _System_Manager_Drone_Operation_Mode{BOOT, DISARMED, GROUND_OPS, FLIGHT, FATAL_FAILURE = -1};
+
+    enum _System_Manager_Cycle_Status {COMPLETED_CYCLE = 0, IN_CYCLE, FAILURE_MODE};
+}
+
+class SystemManager {
+public:
+    SystemManager();
+    void execute();
+    void setState(SystemState& newState);
+private:
+    SystemState* currentState;
+    SM::_System_Manager_Cycle_Status status;
+    SystemManager();
+
+    // TODO PM, AM, TM instances here
+
+    // Mail Queues here
```

Comment:
was the final decision to use mailq?

Reply:
Decision is to go with MailQ until we have something working. It is listed as high priority in the "Post MVP Plans" page. Just don't want to block other development when we have something that works well enough in the mean time.